diff --git a/node_modules/@payloadcms/ui/dist/elements/RenderCustomComponent/index.js b/node_modules/@payloadcms/ui/dist/elements/RenderCustomComponent/index.js
index 52d57c6..64ef4ed 100644
--- a/node_modules/@payloadcms/ui/dist/elements/RenderCustomComponent/index.js
+++ b/node_modules/@payloadcms/ui/dist/elements/RenderCustomComponent/index.js
@@ -1,3 +1,4 @@
+import React from 'react';
 /**
  * Renders a CustomComponent or a Fallback component.
  * Only fallback if the Custom Component is undefined.
@@ -12,6 +13,15 @@
   CustomComponent,
   Fallback
 }) {
-  return CustomComponent !== undefined ? CustomComponent : Fallback;
+  if (CustomComponent === undefined) {
+    return Fallback;
+  }
+  if (CustomComponent === null) {
+    return null;
+  }
+  if (typeof CustomComponent === 'object' && !Array.isArray(CustomComponent) && !React.isValidElement(CustomComponent)) {
+    return Fallback;
+  }
+  return CustomComponent;
 }
 //# sourceMappingURL=index.js.map
diff --git a/node_modules/@payloadcms/ui/dist/forms/fieldSchemasToFormState/renderField.js b/node_modules/@payloadcms/ui/dist/forms/fieldSchemasToFormState/renderField.js
index afb9616..ba86494 100644
--- a/node_modules/@payloadcms/ui/dist/forms/fieldSchemasToFormState/renderField.js
+++ b/node_modules/@payloadcms/ui/dist/forms/fieldSchemasToFormState/renderField.js
@@ -91,7 +91,7 @@ export const renderField = ({
             return;
           }
           row.lastRenderedPath = rowPath;
-          if (fieldConfig.admin?.components && 'RowLabel' in fieldConfig.admin.components) {
+          if (fieldConfig.admin?.components?.RowLabel) {
             if (!row.customComponents) {
               row.customComponents = {};
             }
@@ -122,7 +122,7 @@ export const renderField = ({
           row.lastRenderedPath = rowPath;
           const blockTypeToMatch = row.blockType;
           const blockConfig = req.payload.blocks[blockTypeToMatch] ?? (fieldConfig.blockReferences ?? fieldConfig.blocks).find(block => typeof block !== 'string' && block.slug === blockTypeToMatch);
-          if (blockConfig.admin?.components && 'Label' in blockConfig.admin.components) {
+          if (blockConfig.admin?.components?.Label) {
             if (!fieldState.rows[rowIndex]?.customComponents) {
               fieldState.rows[rowIndex].customComponents = {};
             }
@@ -230,7 +230,7 @@ export const renderField = ({
       }) : 'Mock';
     }
     if (fieldConfig.admin?.components) {
-      if ('afterInput' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.afterInput) {
         fieldState.customComponents.AfterInput = !mockRSCs ? RenderServerComponent({
           clientProps,
           Component: fieldConfig.admin.components.afterInput,
@@ -239,7 +239,7 @@ export const renderField = ({
           serverProps
         }) : 'Mock';
       }
-      if ('beforeInput' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.beforeInput) {
         fieldState.customComponents.BeforeInput = !mockRSCs ? RenderServerComponent({
           clientProps,
           Component: fieldConfig.admin.components.beforeInput,
@@ -248,7 +248,7 @@ export const renderField = ({
           serverProps
         }) : 'Mock';
       }
-      if ('Description' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.Description) {
         fieldState.customComponents.Description = !mockRSCs ? RenderServerComponent({
           clientProps,
           Component: fieldConfig.admin.components.Description,
@@ -257,7 +257,7 @@ export const renderField = ({
           serverProps
         }) : 'Mock';
       }
-      if ('Error' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.Error) {
         fieldState.customComponents.Error = !mockRSCs ? RenderServerComponent({
           clientProps,
           Component: fieldConfig.admin.components.Error,
@@ -266,7 +266,7 @@ export const renderField = ({
           serverProps
         }) : 'Mock';
       }
-      if ('Label' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.Label) {
         fieldState.customComponents.Label = !mockRSCs ? RenderServerComponent({
           clientProps,
           Component: fieldConfig.admin.components.Label,
@@ -275,7 +275,7 @@ export const renderField = ({
           serverProps
         }) : 'Mock';
       }
-      if ('Field' in fieldConfig.admin.components) {
+      if (fieldConfig.admin.components.Field) {
         fieldState.customComponents.Field = !mockRSCs ? /*#__PURE__*/_jsx(WatchCondition, {
           path: path,
           children: RenderServerComponent({
